backend/src/main/java/org/eclipse/tractusx/puris/backend/stock/logic/dto/itemstocksamm/OrderPositionReference.java

```diff
@@ -0,0 +1,80 @@
+/*
+ * Copyright (c) 2023 Volkswagen AG
+ * Copyright (c) 2023 Fraunhofer-Gesellschaft zur Foerderung der angewandten Forschung e.V.
+ * (represented by Fraunhofer ISST)
+ * Copyright (c) 2023 Contributors to the Eclipse Foundation
+ *
+ * See the NOTICE file(s) distributed with this work for additional
+ * information regarding copyright ownership.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Apache License, Version 2.0 which is available at
+ * https://www.apache.org/licenses/LICENSE-2.0.
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+package org.eclipse.tractusx.puris.backend.stock.logic.dto.itemstocksamm;
+
+import com.fasterxml.jackson.annotation.JsonCreator;
+import com.fasterxml.jackson.annotation.JsonProperty;
+import jakarta.validation.constraints.NotNull;
+import lombok.Getter;
+import lombok.Setter;
+
+import java.util.Objects;
+
+/**
+ * Generated class for Reference to Order Position. Encapsulates the references
+ * to identify a position within an order.
+ */
+
+@Getter
+@Setter
+public class OrderPositionReference {
```

Comment:
IMHO all classes in the org.eclipse.tractusx.puris.backend.stock.logic.dto.samm package will be obsolete in the foreseeable future, when will will have moved on to passing itemstocks instead of product stocks. 

And the new implementation has some advantages over the older one: No Optionals, but nullable members. And cleaner structure (i.e less redundant boiler plate code).